File: 1603-design-parking-system/1603-design-parking-system.py
class ParkingSystem:
    def __init__(self, big: int, medium: int, small: int):
        self.Pbig=big
        self.Pmid=medium
        self.Psmall=small

    def addCar(self, carType: int) -> bool:
        if carType<1 or carType>3:
            return False
        
        match carType:
            case 1:
                if self.Pbig>=1:
                    self.Pbig-=1
                    return True
            case 2:
                if self.Pmid>=1:
                    self.Pmid-=1
                    return True
            case 3:
                if self.Psmall>=1:
                    self.Psmall-=1
                    return True
        return False
        


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: 1603-design-parking-system/1603-design-parking-system.cs
public class ParkingSystem {

    int Big {get; set;}
    int Medium {get; set;}
    int Small {get; set;}

    public ParkingSystem(int big, int medium, int small) {
        Big= big;
        Medium=medium;
        Small=small;
    }
    
    public bool AddCar(int carType) {
        if (carType<1 || carType >3){
            return false;
        }

        switch(carType){
            case 1:
            if (Big<=0)
            return false;
            Big-=1;
            break;
            case 2:
            if (Medium<=0)
            return false;
            Medium-=1;
            break;
            case 3:
            if (Small<=0)
            return false;
            Small-=1;
            break;
        }
        return true;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj.AddCar(carType);
 */